src/routers/handlers/check_your_answers/check.your.answers.ts

```diff
@@ -47,9 +51,23 @@ export class CheckYourAnswersHandler extends GenericHandler {
         }
 
         const transactionService = new TransactionService(req.session);
-        await transactionService.closeTransaction();
+        const paymentUrl: string | undefined = await transactionService.closeTransaction();
 
-        return PrefixedUrls.CONFIRMATION;
+        if (!paymentUrl) {
+            logger.debug(`No payment url ${paymentUrl} from closeTransaction, journey redirected to confirmation page`);
+            return PrefixedUrls.CONFIRMATION;
+        } else {
+            // Payment required, start the payment journey
+            logger.debug(`Received payment url ${paymentUrl} from closeTransaction, payment journey started`);
+            const paymentResponse: ApiResponse<Payment> = await startPaymentsSession(req.session, paymentUrl,
+                                                                                     must(getAccountsFilingId(req.session)), must(getTransactionId(req.session)));
```

Comment:
Not sure I like this format, maybe
```suggestion
            const paymentResponse: ApiResponse<Payment> = await startPaymentsSession(req.session, paymentUrl,
                    must(getAccountsFilingId(req.session)), must(getTransactionId(req.session)));
```

Reply:
Are you OK with the below format?